packages/flutter/lib/src/painting/text_painter.dart

```diff
@@ -600,15 +600,15 @@ class TextPainter {
 
   // Complex glyphs can be represented by two or more UTF16 codepoints. This
   // checks if the value represents a UTF16 glyph by itself or is a 'surrogate'.
-  bool _isUtf16Surrogate(int value) {
+  static bool _isUtf16Surrogate(int value) {
     return value & 0xF800 == 0xD800;
```

Comment:
I've opened a separate issue for this and assigned myself.  I'll look into it and merge a small PR separately if it is indeed wrong.  https://github.com/flutter/flutter/issues/59513

Reply:
The test is sufficient. The check uses `& 0xF800` which maps every value in the range 0xD800 .. 0xDFFF to 0xD800.

Not sure if it would be more readable as `value & ~0x7FF == 0xD800` or `value >>11 == 0x1B`.

Reply:
Thanks for the clarification. I'll update the documentation on https://github.com/flutter/flutter/pull/53381#discussion_r441495677.